src/main/webapp/data-visualization.js

```diff
@@ -312,16 +318,34 @@ function setStyle(isCountyQuery) {
   }
 }
 
+// Changes the color of the current visualizations on the page.
+function changeColor(colorParam) {
+  const color = colorParam.value;
+  localStorage.setItem('color', color);
+  // Get variables out of cache
+  const cacheMapsData = JSON.parse(localStorage.getItem('mapsData'));
+  const cacheGeoData = JSON.parse(localStorage.getItem('geoData'));
+  const cacheAmCharts = JSON.parse(localStorage.getItem('amChartsData'));
+  const cacheLocation = localStorage.getItem('location');
+  const cacheDescription = localStorage.getItem('description');
+  // map is undefined on a state query, so check to be sure that 
+  // it is undefined before calling displayCountyGeoJson.
+  if (typeof cacheMapsData !== 'undefined') {
```

Comment:
instead you should call a function that is called both here and in displayVisualization. Here you get values from local storage and then call the function whereas in displayVisualization you retreive values from backend, store them in local storage and call the same function.

Reply:
Do you mean one function to put values in the cache and one function to fetch values from the cache? I understand where I'm supposed to call the function but I'm not really sure what the function does